### PR TITLE
`azurerm_synapse_workspace` - remove sql password requirement for synapse

### PR DESCRIPTION
**Pull request by RaphHaddad**
Please excuse any major mistakes. This is my first PR.

This PR removes the requirement for a local SQL User on Synapse as per Azure's constraints.

I am not sure how to add a validation to check either AAD auth or local SQL user enabled. Any direction is appreciated.

### CODE REVIEW

File: internal/services/synapse/synapse_workspace_resource.go [hunk 1]
```diff
@@ -73,14 +73,14 @@ func resourceSynapseWorkspace() *pluginsdk.Resource {
 
 			"sql_administrator_login": {
 				Type:         pluginsdk.TypeString,
-				Required:     true,
+				Required:     false,
```
Comment (stephybun) [inline internal/services/synapse/synapse_workspace_resource.go:76]:
```suggestion
				Optional:     true,
```

File: internal/services/synapse/synapse_workspace_resource.go [hunk 2]
```diff
@@ -73,14 +73,14 @@ func resourceSynapseWorkspace() *pluginsdk.Resource {
...
 				ForceNew:     true,
 				ValidateFunc: validate.SqlAdministratorLoginName,
 			},
 
 			"sql_administrator_login_password": {
 				Type:      pluginsdk.TypeString,
-				Required:  true,
+				Required:  false,
```
Comment (stephybun) [inline internal/services/synapse/synapse_workspace_resource.go:83]:
```suggestion
				Optional:  true,
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @RaphHaddad! When properties are not required rather than setting the property `Required` to `false` we set `Optional` to `true`.

We would also still need to add a test to validate that this change works, you can probably just copy one of the existing tests and remove the values for `sql_administrator_login` and `sql_administrator_login_password` in the test config. Once that's done we can run the tests and get this merged.

Thanks!

**Comment by RaphHaddad** [comment]
Hello @stephybun, thanks for the guidance.

I have made the changes necessary and added a new test. Can you confirm that the naming is correct for the test?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @RaphHaddad the tests are passing 🎉 . I forgot to mention the docs need to be updated to reflect this change too, but once that's done this will be good to go!

**Comment by RaphHaddad** [comment]
@stephybun , done!

**Review by stephybun — APPROVED** [review]
Thanks for this @RaphHaddad LGTM 🌻

**Pull request merged**